common/config/rush/command-line.json

```diff
@@ -1,41 +1,195 @@
+/**
```

Comment:
Comments in this and other config files were auto-generated by `rush init --overwrite-existing` then manually merged with the existing file contents

Reply:
can we remove them?

Reply:
unless JSON.parse has a mode to ignore them...

Reply:
@dzearing Why do you dislike comments? We spent a lot of time documenting all the settings to make the config file easier for people to understand.

Reply:
@dzearing Why do you dislike comments? We spent a lot of time documenting all the settings to make the config file easier for people to understand.

Reply:
I *LOVE* comments... It's just that json does not support them. Sort of like if you wanted to add comments to your jpg. Sure, you could run a parser to remove them just to read the file...

Reply:
Shouldn't these just be available in the schema anyway tho? Does json schema not give you intellisense?

Reply:
As mentioned elsewhere, the same logic applies to package.json. I would *LOVE* to add comments next to each dependency as to why we have it. But that would break json format.

Reply:
I think having comments in Rush config files is more reasonable than having them in package.json, because Rush itself expects its config files to have comments and will therefore be able to parse them just fine. I also updated our scripts which parse rush.json to strip comments before parsing. Whereas with package.json, it's still likely that a lot of tools (or maybe npm/pnpm themselves, not sure) won't be able to parse comments.

Reply:
To put it another way--Rush is designed to allow comments in its config files. If our custom parsing doesn't support that, we're the ones doing it wrong.

Reply:
I wasn't suggesting to put comments in package.json. I was saying that the logic of not including comments in package.json is the same here. 

I'd like to keep the file parsable, so that we can write tools that can read (or generate) the file without worrying that JSON.parse won't work.

Reply:
an alternative is that we abandon all usage of JSON.parse and use https://www.npmjs.com/package/jsonc-parser or something instead. 

Pete mentioned below a helper to read rush.json, but it's unclear to me if it has additional benefits over just using jsonc-parser.

Reply:
I know you weren't suggesting putting comments in package.json, I just meant that the comparison isn't quite valid because tooling around package.json is not designed to handle comments, whereas tooling within Rush *is* designed that way.

According to Pete, the way you're supposed to parse Rush config files is not by using JSON.parse, it's by using `@microsoft/rush-lib` (which will handle comments). I'm hesitant to pull in rush-lib since it's a rather large library (depends on node-core-library too), but if we're going to roll our own parsing, it really ought to support the same features as the official parsing.

Reply:
I could also switch to using rush-lib for parsing, I was just hesitant to do that like I mentioned because it's a big library (and I was under the impression the reason we switched to a custom toolchain awhile back was to reduce the number of dependencies and decrease install and build time)

Reply:
No modern tool should be using `JSON.parse()` or `require()` to parse a config file.  These APIs cannot validate JSON schemas, and they generally don't provide accurate line numbers or helpful error messages.  `require()` has a security issue that it will eval any code that it finds in the data file.  Also `require()` assumes that JSON is a proper subset of JavaScript, which is [technically incorrect](http://timelessrepo.com/json-isnt-a-javascript-subset);

We use **@microsoft/node-core-library** for general config filers, and **@microsoft/rush-lib** for reading Rush's specific config files.  But really any modern library such as **jsonc-parser** will support comments just fine.

(I do find it absolutely annoying that GitHub's syntax highlighter shows comments as bright red errors.  It looks really bad, given that the vast majority of JSON files are config files, and it's a complete nonstarter for a professional tool to have config files that cannot be documented.  If we can't get GitHub to fix their syntax highlighter, I suppose we could consider renaming all of Rush's .json file extensions to .jsonc or .json5. That's a nontrivial change, however.)